### PR TITLE
Feature map keys

### PR DESCRIPTION
**Pull request by eclipxe13**
Declare `MapInterface::keys` method that mimics java `Map::keySet`, implement it on `AbstractMap` and create tests.

### CODE REVIEW

File: src/Map/AbstractMap.php [hunk 1]
```diff
@@ -44,6 +44,12 @@ public function containsValue($value)
         return in_array($value, $this->data, true);
     }
 
+    public function keys()
+    {
+        return array_keys($this->data);
+    }
+
+
```
Comment (ramsey) [inline src/Map/AbstractMap.php]: Nitpicky: get rid of the extra newline.

**Comment by ramsey** [comment]
Looks great. Get rid of that extra newline in `src/Map/AbstractMap.php` and make sure `.gitignore` has a trailing newline, and I'll merge it in.

Thanks!

**Comment by eclipxe13** [comment]
done :)

**Comment by ramsey** [comment]
Thanks!

**Pull request merged**